libsolidity/analysis/DeclarationContainer.h

```diff
@@ -58,11 +58,17 @@ class DeclarationContainer
 	/// @returns whether declaration is valid, and if not also returns previous declaration.
 	Declaration const* conflictingDeclaration(Declaration const& _declaration, ASTString const* _name = nullptr) const;
 
+	/// @returns existing declaration names similar to @a _name.
+	std::vector<ASTString> similarNames(ASTString const& _name) const;
+
+
 private:
 	ASTNode const* m_enclosingNode;
 	DeclarationContainer const* m_enclosingContainer;
 	std::map<ASTString, std::vector<Declaration const*>> m_declarations;
 	std::map<ASTString, std::vector<Declaration const*>> m_invisibleDeclarations;
+
+	static size_t const MAXIMUM_DISTANCE = 2;
```

Comment:
Not fully sold we should have this constant here, but if we the name should reflect what it does.

Reply:
I agree, it should just be a constant inside the `similarNames` function.